platform/darwin/src/MGLOfflineStorage.h

```diff
@@ -448,6 +448,37 @@ MGL_EXPORT
 
 - (void)putResourceWithUrl:(NSURL *)url data:(NSData *)data modified:(nullable NSDate *)modified expires:(nullable NSDate *)expires etag:(nullable NSString *)etag mustRevalidate:(BOOL)mustRevalidate __attribute__((deprecated("", "-preloadData:forURL:modificationDate:expirationDate:eTag:mustRevalidate:")));
 
+
+/**
+A block to be called once the data  has been preloaded.
+*/
+typedef void (^MGLOfflinePreloadDataCompletionHandler)(void);
```

Comment:
Do you think there’s a potential for the preloading operation to fail with an error, either now or in a future change? If so, let’s add an NSError-typed error parameter here to avoid a backwards-incompatible change.

Reply:
done. Completion handler has url and an error.